### DISCUSSION TITLE
Erosion in /MAT/LAW24 + subsequent /INTER/TYPE24 update

### DISCUSSION THREAD

**Discussion by torefindsen** [OpenRadioss/OpenRadioss#3003] [Q&A]
Hi all, 

I'm trying to model an embedded steel hook being extracted from a fixed block of concrete, as seen in the attached sketch - I plan to model the concrete using /MAT/LAW24 - I face 2 problems:

1. How do I remove (erode?) failed elements in the concrete? (It seems I need a failure model /FAIL, but only COCKCROFT,ENERGY,TENSSTRAIN and TBUTCHER are compatible with MAT/LAW24, and neither seems well-suited?)

2. After removing/eroding elements, how do I update the contact definition for /INTER/TYPE24, such that contact between hook and the "newly" exposed concrete elements is established?  

![hook_embedded_in_concrete](https://github.com/user-attachments/assets/fcc845b0-fac9-4b89-8b82-2c7c76a0840e)

Thanks in advance!
Br Tore

**Comment by LYS0922** [comment]
hi, even if I did not use this material law , I would like to give you another method.
You could solve this problem by using user subroutine .
there is a variable you could use to control the element removing.
by this way ,you could let an element die or not by the conditions you want.

**Reply by torefindsen** [reply]
Hi @LYS0922 - thanks for your suggestion - I would like to pursue "built-in" functionality before resorting to user sub-routines, but I will certainly keep your suggestion in mind!
Br
Tore

**Comment by RevolT87** [comment]
Hi @torefindsen,
1- element erosion:
Directly in the material law 24 you have the parameter epsilon_max which triggers element erosion depending on total failure strain.
You may also use quality or time step criteria to trigger element erosion. These parameters can be find in the /PROP/SOLID card:
![Capture d'écran 2024-09-23_164045](https://github.com/user-attachments/assets/831e57b6-865c-4e4f-a22c-e14a62f1d1bd)

2- to consider element erosion in contacts, you may use the flag Idel=1 in the type24 interface (same with type7, 11, 19, 25,..)

3- FYI, there is also a new concrete material law you my want to try: 
https://community.altair.com/community/en/radioss-validation-study-johnson-holmquist-concrete-material-law?id=kb_article_view&sysparm_article=KB0124985.

I hope it helps,
Regards,
Mathis

**Reply by torefindsen** [reply]
Hi @RevolT87, thanks for your suggestions - 

ad 1) I have now played a bit with the element erosion criteria in /PROP/SOLID and while I can now enable erosion, my model is still highly unstable, I suspect this is related to the contact interface as per 2

ad 2) I currently have 2 "static" surface sets, defined prior to the analysiss, comprising the 2 sides of my interface - I assume I must use another methodology as the elements attached to these nodes are removed during erosion - ***should I use use /SURF/PART/ALL and put ALL nodes in the part that erodes in the secondary-node set?*** 

ad 3) Thanks, MAT/LAW126 indeed looks very interesting, for now I'm drawn toward MAT/LAW24 for the ability to only define f_c and have the remaining failure parameters estimated from that 

Br
Tore

**Reply by RevolT87** [reply]
Hi @torefindsen,
sorry for the late answer. 
For the contact interface, I would recommand to use /INTER/TYPE7 with /SURF/PART/ALL, Idel=1 and a small gapmin value.
For the stability topic, which element formulation are you using? Do you have a lot of Houglass energy?

Regards,
Mathis.
